Lab_7_Ледовская Мария_БИВТ-24-16

ldvsked opened the pull request.

NemirovichMark committed:
Update README.md

ldvsked committed:
Add files via upload

ldvsked committed:
Add files via upload

ldvsked committed:
Add files via upload

ldvsked committed:
Delete Purple-1.cs

NemirovichMark commented:
Здравствуйте, Мария.

Описание ошибок можно найти в словаре:
https://newlms.misis.ru/mod/glossary/view.php?id=65847&mode=letter&hook=SPECIAL
Выгрузка системы тестирования в файле (обращайте внимание на первую строку с подсказкой, где ошибка (должно начинаться с Lab_7, но не с Lab_7_Test):
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19428562/Lab_7_log_Purple.txt)

ldvsked committed:
Здравствуйте, 1,2,3 исправила, к 4 и 5 вопросы

В 4 правда не понимаю, что может пойти не так, почему ответ неправильный, в 5 добавила проверку, что _researchId не может быть меньше 1, если не поможет, тогда тоже не знаю, что не так

ldvsked committed:
Add files via upload

NemirovichMark commented:
Здравствуйте, Мария.

Запускайте работу перед отправкой, пожалуйста, чтобы таких ошибок не было.
![image](https://github.com/user-attachments/assets/8a5457ba-679f-4cec-8905-c9ee8546f3a6)
И имя свойства поправить
![image](https://github.com/user-attachments/assets/cc06c61b-b3ed-430c-bfd2-1d3e615470c5)


[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19453036/Lab_7_log_Purple.txt)

ldvsked committed:
Исправила 1,2 опечатки

NemirovichMark commented:
Хорошо 👍 
Остальные ошибки описаны в файле под скринами.

ldvsked committed:
Add files via upload

NemirovichMark commented:
Здравствуйте, Мария.

[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19502022/Lab_7_log_Purple.txt)

ldvsked committed:
Добрый вечер, попыталась исправить, вопрос по 4 

Подскажите пожалуйста почему метод shuffle некорректно работает в purple_4. В упор не понимаю где может быть ошибка

NemirovichMark commented:
Здравствуйте, Мария.

[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19594035/Lab_7_log_Purple.txt)

ldvsked committed:
Исправила 1 и 2

NemirovichMark commented:
Здравствуйте, Мария.
Неверно оценки выставляете в 3м
![image](https://github.com/user-attachments/assets/55a43205-09d2-4dc4-bf65-b6a8876ea12a)
В 4м
![image](https://github.com/user-attachments/assets/42dd1d25-8c05-456e-b172-cc01e2fd2067)

В 5м
![image](https://github.com/user-attachments/assets/a0231df9-801c-421d-9c56-df5b8bac67da)

ldvsked committed:
Исправила 3  и 5, вопрос по 4 в тимсе

NemirovichMark commented:
Здравствуйте, Мария.
В 3м ошибка в:
                //ModificateMood();
А как раскомментируете, будет в
  Array.Sort(marks_per_judge);
  Array.Reverse(marks_per_judge);
По 4му объяснил в Тимсе. В задании дан пример последовательности, далее указание, кто её начинает.
В 5м ошибка в:
                _researches[_researchId-1]=newResearch;
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19651279/Lab_7_log_Purple.txt)

ldvsked committed:
Исправила 4 и мб 5

ldvsked committed:
Исправила 3 с реверсом

NemirovichMark commented:
Здравствуйте, Мария.
В 3м номере да реверс нужен был, но эта комбинация: sort + reverse - неустойчивая сортировка
![image](https://github.com/user-attachments/assets/30effe31-d7fc-405d-8114-754227f947d6)
Поменять знак
 if (participants[j]._places[6] > participants[j + 1]._places[6])

Куда проще можно в цикле сделать: отсортировать по оценкам у i-го судьи, проставить места участникам. И все. Они как раз окажутся отсортированными по последнему судье в конце ( я это условие добавил, чтобы упростить вам задание, а получилось, двойную работу делаете))
KISS - keep is simple, student ;)

В 5м номере добавить проверку на корректный вопрос.
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19664467/Lab_7_log_Purple.txt)

ldvsked committed:
3 попроще переписала и в 5 проверку на question

NemirovichMark commented:
3й Score = 0 всегда
5й 👍 
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19665370/Lab_7_log_Purple.txt)

ldvsked committed:
Forgot about counting score

NemirovichMark commented:
Здравствуйте, Мария.
Index was outside the bounds of the array.
in C:\Users\Markosias\Documents\BIVT-24\BIVT-2024-Lab-7-test\Lab_7\Purple_3.cs:line 233
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19680636/Lab_7_log_Purple.txt)

ldvsked committed:
исправила 3

NemirovichMark commented:
Здравствуйте, Мария.
Отлично! Работа принята. Допуск к защите выдан.
Загрузите файлы на Moodle и приходите на защиту.
[Lab_7_log_Purple.txt](https://github.com/user-attachments/files/19773208/Lab_7_log_Purple.txt)
